Mathlib/LinearAlgebra/Eigenspace/Charpoly.lean

```diff
@@ -0,0 +1,34 @@
+/-
+Copyright (c) 2025 Lawrence Wu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Lawrence Wu
+-/
+import Mathlib.LinearAlgebra.Charpoly.ToMatrix
+import Mathlib.LinearAlgebra.Eigenspace.Matrix
+import Mathlib.LinearAlgebra.Matrix.Charpoly.Eigs
+
+/-!
+# Eigenvalues are the roots of the characteristic polynomial.
+
+## Tags
+
+eigenvalue, characteristic polynomial
+-/
+
+namespace Module
+
+namespace End
+
+open LinearMap
+
+variable {K V : Type*} [Field K] [AddCommGroup V] [Module K V] [Module.Finite K V]
```

Comment:
Author's note: Going through `hasEigenvalue_iff_mem_spectrum` (which requires `Field`) actually loses generality. I have a proof of the following:

```lean
lemma hasEigenvalue_iff_isRoot_charpoly_of_isDomain {K V : Type*} [CommRing K] [IsDomain K] [AddCommGroup V] [Module K V] [Module.Free K V] [Module.Finite K V] (f : Module.End K V) (μ : K) :
    f.HasEigenvalue μ ↔ f.charpoly.IsRoot μ := by
  sorry
```

and will clean it up and add it after I determine the necessity of `IsDomain`